src/NuGet.Core/NuGet.Commands/RestoreCommand/ProjectRestoreCommand.cs

```diff
@@ -76,20 +76,7 @@ public ProjectRestoreCommand(ProjectRestoreRequest request)
 
             telemetryActivity.EndIntervalMeasure(telemetryPrefix + WalkFrameworkDependencyDuration);
 
-            telemetryActivity.StartIntervalMeasure();
-
-            var downloadDependencyResolutionTasks = new List<Task<DownloadDependencyResolutionResult>>();
-            foreach (var targetFrameworkInformation in _request.Project.TargetFrameworks)
-            {
-                downloadDependencyResolutionTasks.Add(ResolveDownloadDependenciesAsync(
-                    context,
-                    targetFrameworkInformation,
-                    token));
-            }
-
-            var downloadDependencyResolutionResults = await Task.WhenAll(downloadDependencyResolutionTasks);
-
-            telemetryActivity.EndIntervalMeasure(telemetryPrefix + EvaluateDownloadDependenciesDuration);
+            var downloadDependencyResolutionResults = await DownloadDependenciesAsync(_request.Project, context, telemetryActivity, telemetryPrefix, token);
```

Comment:
This functionality was abstracted out to be a resusable method so both code paths could call it.